timelock-impl/src/main/java/com/palantir/atlasdb/timelock/TimelockNamespaces.java

```diff
@@ -63,8 +68,27 @@ public TimelockNamespaces(
         registerClientCapacityMetrics(metrics);
     }
 
-    public TimeLockServices get(String namespace) {
-        return services.computeIfAbsent(namespace, this::createNewClient);
+    public static Optional<String> toUserAgent(@Nullable RequestContext context) {
```

Comment:
happy to just merge this into the below method. Originally I was expecting we'd handle conversions for non-undertow interfaces too. I was also hoping we'd be able to extract a UserAgent rather than just its string as well, and so was trying to keep the conjure RequestContext out of the main getter here. But given all production code should have this type it doesn't really need to be a separate helper method that clients call first